frontend/src/modules/organization/components/edit/domain/organization-domain-add.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <lf-modal v-model="isModalOpen">
+    <template #default="{ close }">
+      <div class="px-6 pt-4 pb-10">
+        <div class="flex items-center justify-between pb-6">
+          <h5>Add {{ modelValue?.type?.replace('-', ' ') }}</h5>
+          <lf-button type="secondary-ghost-light" :icon-only="true" @click="close">
+            <lf-icon name="close-line" />
+          </lf-button>
+        </div>
+
+        <div class="flex flex-col gap-3 items-start">
+          <article v-for="(domain, di) of form" :key="di" class="w-full">
+            <lf-field class="w-full">
+              <div class="flex items-center w-full">
+                <lf-input
+                  v-model="domain.value"
+                  class="h-10 flex-grow"
+                  :class="showVerified ? '!rounded-r-none' : ''"
+                  placeholder="Enter URL"
+                  :invalid="$v.form[di].value.$invalid && $v.form[di].value.$dirty"
+                  @blur="$v.form[di].value.$touch()"
+                >
+                  <template #prefix>
+                    <lf-icon name="link" class="text-black" :size="20" />
+                  </template>
+                </lf-input>
+                <label v-if="showVerified" class="border border-gray-200 h-10 py-2.5 px-3 border-l-0 cursor-pointer rounded-r-lg">
+                  <lf-checkbox v-model="domain.verified" class="!flex-nowrap">
+                    Verified
+                  </lf-checkbox>
+                </label>
+                <lf-button
+                  v-if="form.length > 1"
+                  class="ml-3"
+                  type="secondary-ghost-light"
+                  :icon-only="true"
+                  @click="form.splice(di, 1)"
+                >
+                  <lf-icon name="delete-bin-6-line" />
+                </lf-button>
+              </div>
+              <lf-field-messages
+                :validation="$v.form[di].value"
+                :error-messages="{
+                  required: 'This field is required',
+                }"
+              />
+            </lf-field>
+          </article>
+          <lf-button
+            type="primary-link"
+            size="small"
+            @click="form.push({
+              ...defaultForm,
+              ...props.modelValue,
+            })"
+          >
+            <lf-icon name="add-line" />
+            Add domain
+          </lf-button>
```

Comment:
We might need to add logic here so that if user is trying to add 2 domains in a scenario where the organization doesn't have any verified primary domains, the verified options next to the domain are disabled if one verified option is already selected.
![Screenshot 2024-08-05 at 16 05 01](https://github.com/user-attachments/assets/5f910f8b-b162-4300-866e-1bb647531dde)